der-oid-macro/src/lib.rs

```diff
@@ -123,17 +123,18 @@ pub fn oid(item: TokenStream) -> TokenStream {
     s.push(']');
 
     let code = if raw {
-        s
+        format!("{{extern crate alloc; {}}}", s)
     } else if relative {
         format!(
-            "der_parser::oid::Oid::new_relative(std::borrow::Cow::Borrowed(&{}))",
+            "{{extern crate alloc; der_parser::oid::Oid::new_relative(alloc::borrow::Cow::Borrowed(&{}))}}",
```

Comment:
Another solution would be to re-export `alloc` from `der-parser`, hiding it in documentation, and use `der-parser::export::alloc` (see https://github.com/rusticata/rusticata-macros/commit/cbb3117a478c0980b4de5f6dd1700f3433a14a5d for an example). This way, the caller does not even have to declare it in `Cargo.toml`.